### PR TITLE
Add support for software Pulse-width Modulation (PwM) on all channels

### LINKED ISSUE #3: Pulse width Modulation (PwM)
This will probably require modifications to the kernel

### LINKED ISSUE #4: There are not many examples and some are out of date

### LINKED ISSUE #13: Move TODOs out of RPi/_GPIO.py comments

### PR DESCRIPTION
**Pull request by theyoyojo**
```
summary of updates in this patch:
	pwm:
	- implement PwM interface as per spec
	- implement basic software PwM functionality
	- add logic tests for PwM interface
	- extend threading object hierarchy to integrate PwM thread
		control with existing thread control flow logic
		this includes remaming _PollThread as _LineThread
	- add _line_thread_* thread type constants
	- add thread_{stop,start} interface to _Line to avoid direct
		manipulation of _LineThread
	- add dutycycle and frequency fields to class _Line
	- update spec to reflect new internal changes (in line_* interface)
	- bump spec revision number to 2
	- s/line_start_poll/line_poll_start/g
	- add dictionary mapping thread types to thread entry points

	examples:
	- fix broken examples
	- add flashing light example
	- deprecate morese code example
	- remove old wrap.py prototype
	- fix and enforce style in examples
	- set executable bit on all examples

	misc:
	- add a note about notes.sh to README
	- add header to RPi/_GPIO
	- mention that the project is free software in the README and header
	- export the PWM name in RPi/GPIO/__init__.py for access via GPIO name
	- s/is_all_bools/is_all_bools_or_directions/g
		this function now returns True when GPIO.HIGH or GPIO.LOW
		are the data or in the list
	- fix GPIO.HIGH and GPIO.LOW all being interpreted as True by output()
	- remove the directory named 'misc' that contained files
		to reproduce a random bug and should not have been
		in the master branch
	- fix test-style.sh not scanning tests and examples
	- fix a lot of style errors in the tests
	- replace some magic numbers with hopefully more descriptive constants
	- remove large ascii art TODO and redundant comments at the top of _GPIO.py

	tl;dr: Anyway, here's software PwM

Signed-off-by: Joel Savitz <joelsavitz@gmail.com>
```

fixes #3 
fixes #4 
fixes #13

### CODE REVIEW

**Comment by theyoyojo** [comment]
obsoleted by #27
